rust/src/measurements/make_private_expr/mod.rs

```diff
@@ -0,0 +1,95 @@
+use opendp_derive::bootstrap;
+use polars_plan::dsl::Expr;
+
+use crate::{
+    core::{Measure, Measurement, Metric, MetricSpace},
+    domains::{ExprDomain, OuterMetric},
+    error::Fallible,
+    measures::MaxDivergence,
+    metrics::{InsertDeleteDistance, PartitionDistance, SymmetricDistance},
+    transformations::{traits::UnboundedMetric, DatasetOuterMetric},
+};
+
+#[cfg(feature = "ffi")]
+mod ffi;
+
+#[bootstrap(
+    features("contrib"),
+    arguments(
+        output_measure(c_type = "AnyMeasure *", rust_type = b"null"),
+        param(rust_type = "Option<f64>", c_type = "AnyObject *", default = b"null")
+    ),
+    generics(MI(suppress), MO(suppress))
+)]
+/// Create a differentially private measurement from an [`Expr`].
+///
+/// # Arguments
+/// * `input_domain` - The domain of the input data.
+/// * `input_metric` - How to measure distances between neighboring input data sets.
+/// * `output_measure` - How to measure privacy loss.
+/// * `expr` - The [`Expr`] to be privatized.
+/// * `param` - A tune-able parameter that affects the privacy-utility tradeoff.
+pub fn make_private_expr<MI: 'static + Metric, MO: 'static + Measure>(
+    input_domain: ExprDomain,
+    input_metric: MI,
+    output_measure: MO,
+    expr: Expr,
+    param: f64,
+) -> Fallible<Measurement<ExprDomain, Expr, MI, MO>>
+where
+    Expr: PrivateExpr<MI, MO>,
+    (ExprDomain, MI): MetricSpace,
+{
+    expr.make_private(input_domain, input_metric, output_measure, param)
+}
+
+pub trait PrivateExpr<MI: Metric, MO: Measure> {
+    fn make_private(
+        self,
+        input_domain: ExprDomain,
+        input_metric: MI,
+        output_metric: MO,
+        param: f64,
+    ) -> Fallible<Measurement<ExprDomain, Expr, MI, MO>>;
+}
+
+macro_rules! impl_private_expr {
+    ($($MI:ty),*) => {
+        $(impl PrivateExpr<$MI, MaxDivergence<f64>> for Expr {
+            fn make_private(
+                self,
+                _input_domain: ExprDomain,
+                _input_metric: $MI,
+                _output_measure: MaxDivergence<f64>,
+                _param: f64,
+            ) -> Fallible<Measurement<ExprDomain, Expr, $MI, MaxDivergence<f64>>> {
+
+                match self {
+                    _ => fallible!(MakeMeasurement, "Unsupported expr: {:?}", self),
+                }
+            }
+        })*
+    };
+}
+
+impl_private_expr!(SymmetricDistance, InsertDeleteDistance);
```

Comment:
Explain why it only applies to these two distances? Is there something analogous for the others? Maybe that's what the following block is about?

Reply:
Removed for now. It will become useful later.